wordpress/wp-content/plugins/sil-dictionary-webonary/include/searchform_func.php

```diff
@@ -24,6 +24,63 @@ function webonary_searchform() {
 	{
 		return false;
 	}
+
+	/*
+	 * Set up the Parts of Speech
+	 */
+	$taxonomy = filter_input(INPUT_GET, 'tax', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+	$parts_of_speech_dropdown = '';
+
+	if(get_option('useCloudBackend'))
+	{
+		$arrIndexed = array();
+		$dictionaryId = Webonary_Cloud::getBlogDictionaryId();
+		$dictionary = Webonary_Cloud::getDictionary($dictionaryId);
+		if(!is_null($dictionary))
```

Comment:
What if `$dictionary` is indeed null?

Reply:
ah, yes, indeed. the latest commit makes sure that all variables set in this block are properly initialized outside of it, so that no results are shown if the dictionary does not exist.